apps/nextjs-website/src/app/auth/sign-up/page.tsx

```diff
@@ -0,0 +1,132 @@
+'use client';
+import { translations } from '@/_contents/translations';
+import PageNotFound from '@/app/not-found';
+import CheckItem from '@/components/molecules/CheckItem/CheckItem';
+import ConfirmSignUp from '@/components/organisms/Auth/ConfirmSignUp';
+import SignUpForm from '@/components/organisms/Auth/SignUpForm';
+import { environment } from '@/config';
+import { SignUpSteps } from '@/lib/types/signUpSteps';
+import { Box, Grid, Typography, useMediaQuery } from '@mui/material';
+import { Auth } from 'aws-amplify';
+import { useCallback, useState } from 'react';
+
+const SignUp = () => {
+  const {
+    auth: { signUp },
+  } = translations;
+
+  const isSmallScreen = useMediaQuery('(max-width: 1000px)');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [company, setCompany] = useState('');
+  const [role, setRole] = useState('');
+  const [signUpStep, setSignUpStep] = useState(SignUpSteps.SIGN_UP);
+
+  const onSignUp = useCallback(async () => {
+    // TODO: Add company and role to user attributes
+    const result = await Auth.signUp({
+      username,
+      password,
+      attributes: {
+        given_name: firstName,
+        family_name: lastName,
+      },
+    }).catch(() => {
+      return false;
+    });
```

Comment:
The system must handle errors returned by cognito (e.g. Username already exists) showing something to the user. The `signUp` method returns a rejected promise containing details about the errors in the following form:

```
{"name":"UsernameExistsException","code":"UsernameExistsException"}
```

Sadly seems that Amplify [doesn't expose a type with all the errors code](https://github.com/aws-amplify/amplify-js/issues/9104). But all the errors returned by the sign-up method are defined on [aws cognito docs](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_SignUp.html#API_SignUp_Errors).